SCons/Tool/msvs.py

```diff
@@ -1867,7 +1915,22 @@ def GenerateProject(target, source, env):
         GenerateDSW(dswfile, source, env)
 
 def GenerateSolution(target, source, env) -> None:
-    GenerateDSW(target[0], source, env)
+
+    builddswfile = target[0]
+    dswfile = builddswfile.srcnode()
```

Comment:
I don't think we should ever be writing into the src directory of a variant (or into a Repository())
What's the reason for this?

Reply:
Some previous discussion here: https://github.com/SCons/scons/issues/4612#issuecomment-2408697677 and a number of successive posts.

The generated solution and project files have source code references which is why I'm guessing they were originally written the way the are.  Opening the visual studio solution in the source directory will still build in the variant directory.

If the project artifacts were generated in the variant build directory, then all source files would need relative paths back to the source directory.  There are a number of user specified locations that may be affected as well.

The project files have been written this way going way back (possibly to version 1).  While the project file were written with a placeholder in the variant dir, the solution file was generated in the variant dir rather than source folder.

Getting the relative links to the source file locations is not easy.  I was working on attempting the generate the solution/project files in their own folder (i.e., not the source folder and not the variant dir) but didn't know how some of the user specified paths/files should work.  For example, is a file specified with no path because the generated environment locates it?

As one would have one project file may use multiple variant directories (i.e., one build folder for debug and one build folder for release).  There is only one solution file and one project file (with both msvs variants).  If there were two build folders the solution and project files would have to written to multiple folders (i.e., multiple copies).